File: packages/node/examples/basic-usage.ts
/**
 * Example: Basic usage of the Telegram Wallet P2P SDK (Unofficial)
 *
 * Before running:
 *   npm install telegram-wallet-p2p
 *
 * Set your API key:
 *   export WALLET_P2P_API_KEY=your-api-key
 *
 * Run:
 *   npx tsx examples/basic-usage.ts
 *
 * This is an UNOFFICIAL SDK.
 * Not affiliated with Telegram or Wallet.
 */

import {
    WalletP2PClient,
    TradeSide,
    bestPrice,
    priceSpread,
    marketSummary,
} from "telegram-wallet-p2p";

async function main() {
    const apiKey = process.env.WALLET_P2P_API_KEY;
    if (!apiKey) {
        console.error("Error: Set WALLET_P2P_API_KEY environment variable");
        process.exit(1);
    }

    const client = new WalletP2PClient({ apiKey });

    // Fetch USDT/RUB sell ads
    const response = await client.getOnlineItems({
        cryptoCurrency: "USDT",
        fiatCurrency: "RUB",
        side: TradeSide.SELL,
        pageSize: 20,
    });

    console.log(`Found ${response.data.length} ads\n`);

    // Display each ad
    for (const item of response.data) {
        console.log(
            `  [${item.merchantLevel}] ${item.nickname} — ` +
            `₽${item.price}/USDT — ` +
            `Qty: ${item.lastQuantity} — ` +
            `Methods: ${item.payments.join(", ")}`
        );
    }

    // Analytics
    console.log("\n--- Price Spread ---");
    const spread = priceSpread(response.data);
    if (spread) {
        console.log(`  Min: ${spread.min}`);
        console.log(`  Max: ${spread.max}`);
        console.log(`  Avg: ${spread.avg}`);
        console.log(`  Spread: ${spread.spread}`);
    }

    console.log("\n--- Best Price ---");
    const best = bestPrice(response.data, TradeSide.SELL);
    if (best) {
        console.log(`  Best sell: ₽${best.price} by ${best.nickname}`);
    }

    console.log("\n--- Market Summary ---");
    const summary = marketSummary(response.data);
    console.log(`  Total ads: ${summary.totalAds}`);
    console.log(`  Online: ${summary.onlineCount}`);
    console.log(`  Auto-accept: ${summary.autoAcceptCount}`);
    console.log(`  Avg completion rate: ${summary.avgCompletionRate}`);
}

main().catch(console.error);

File: packages/python/examples/basic_usage.py
"""
Example: Basic usage of the Telegram Wallet P2P SDK (Unofficial)

Before running:
  pip install telegram-wallet-p2p

Set your API key as an environment variable:
  export WALLET_P2P_API_KEY=your-api-key

This is an UNOFFICIAL SDK.
Not affiliated with Telegram or Wallet.
"""

import asyncio
import os

from telegram_wallet_p2p import WalletP2PClient, TradeSide
from telegram_wallet_p2p.analytics import best_price, market_summary, price_spread


async def main():
    api_key = os.environ.get("WALLET_P2P_API_KEY")
    if not api_key:
        print("Error: Set WALLET_P2P_API_KEY environment variable")
        return

    async with WalletP2PClient(api_key=api_key) as client:
        # Fetch USDT/RUB sell ads
        response = await client.get_online_items(
            crypto_currency="USDT",
            fiat_currency="RUB",
            side=TradeSide.SELL,
            page_size=20,
        )

        print(f"Found {len(response.data)} ads\n")

        # Display each ad
        for item in response.data:
            print(
                f"  [{item.merchant_level.value}] {item.nickname} — "
                f"₽{item.price}/USDT — "
                f"Qty: {item.last_quantity} — "
                f"Methods: {', '.join(item.payments)}"
            )

        # Analytics
        print("\n--- Price Spread ---")
        spread = price_spread(response.data)
        if spread:
            for key, value in spread.items():
                print(f"  {key}: {value}")

        print("\n--- Best Price ---")
        best = best_price(response.data, TradeSide.SELL)
        if best:
            print(f"  Best sell: ₽{best.price} by {best.nickname}")

        print("\n--- Market Summary ---")
        summary = market_summary(response.data)
        for key, value in summary.items():
            print(f"  {key}: {value}")


if __name__ == "__main__":
    asyncio.run(main())
